### ISSUE TITLE
.pylintrc file is used on OSX but not in Travis CI task

### ISSUE THREAD

**Issue by duhaime** [PyCQA/pylint#1868]
### Steps to reproduce
1. Create custom pylintrc file
2. Create Travis CI task that runs pylint and passes in relative path to pylintrc file
3. Deploy Travis and see that pylintrc file is not passed into pylint process

### Current behavior

I'm working on a project that would like to use pylint on some automated Travis CI builds. I wrote a custom pylintrc file and pass it in to a linting task on OSX (10.12.6) with no issue. However, when I attempt to run the pylint task on Travis (in a Linux container) the lint task does not seem to accept the pylint file.

Here's the pylint result on OSX:

```
> pwd && ls -lsah server && pylint server/app.py --rcfile=./server/.pylintrc

/Users/yaledhlab/Desktop/cats
total 16
0 drwxr-xr-x   6 yaledhlab  staff   204B Feb  5 07:32 .
0 drwxr-xr-x  14 yaledhlab  staff   476B Feb  5 07:32 ..
8 -rw-r--r--@  1 yaledhlab  staff   3.6K Feb  4 16:23 .pylintrc
0 drwxr-xr-x   3 yaledhlab  staff   102B Feb  4 22:11 __pycache__
8 -rw-r--r--   1 yaledhlab  staff   581B Feb  4 22:08 app.py
0 drwxr-xr-x   4 yaledhlab  staff   136B Feb  4 19:19 tests
Using config file /Users/yaledhlab/Desktop/let-them-speak/server/.pylintrc

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

Here's the pylint result on Travis:

```
> pwd && ls -lsah server && pylint server/app.py --rcfile=./server/.pylintrc
/home/travis/build/duhaime/cats
total 8.0K
   0 drwxr-xr-x 3 travis travis   50 Feb  5 13:26 .
   0 drwxr-xr-x 6 travis travis  228 Feb  5 13:27 ..
4.0K -rw-r--r-- 1 travis travis  581 Feb  5 13:26 app.py
4.0K -rw-r--r-- 1 travis travis 3.7K Feb  5 13:26 .pylintrc
   0 drwxr-xr-x 2 travis travis   25 Feb  5 13:26 tests
************* Module app
I: 25, 0: Locally disabling unused-argument (W0613) (locally-disabled)
C:  5, 0: Invalid constant name "app" (invalid-name)
```

Here's the full content of `server/.pylintrc` (in which constant names are expected to be snake_case):

```
# Pylint output cheatsheet:
# [R]efactor for a “good practice” metric violation
# [C]onvention for coding standard violation
# [W]arning for stylistic problems, or minor programming issues
# [E]rror for important programming issues (i.e. most probably bug)
# [F]atal for errors which prevented further processing

[MASTER]

# Use multiple processes to speed up Pylint.
jobs=4

# Only show warnings with the listed confidence levels. Leave empty to show
# all. Valid levels: HIGH, INFERENCE, INFERENCE_FAILURE, UNDEFINED
confidence=


[REPORTS]

# Python expression which should return a note less than 10 (10 is the highest
# note). You have access to the variables errors warning, statement which
# respectively contain the number of errors / warnings messages and the total
# number of statements analyzed. This is used by the global evaluation report
# (RP0004).
evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)

# Set the output format. Available formats are text, parseable, colorized, json
# and msvs (visual studio).You can also give a reporter class, eg
# mypackage.mymodule.MyReporterClass.
output-format=text

# Tells whether to display a full report or only the messages
reports=no

# Activate the evaluation score.
score=yes


[REFACTORING]

# Maximum number of nested blocks for function / method body
max-nested-blocks=5


[BASIC]

# Naming style matching correct argument names
argument-naming-style=snake_case

# Naming style matching correct attribute names
attr-naming-style=snake_case

# Bad variable names which should always be refused, separated by a comma
bad-names=foo,
          bar,
          baz,
          cats

# Naming style matching correct class names
class-naming-style=PascalCase

# Naming style matching correct constant names
const-naming-style=snake_case

# Naming style matching correct function names
function-naming-style=snake_case

# Naming style matching correct module names
module-naming-style=snake_case

# Naming style matching correct variable names
variable-naming-style=snake_case


[FORMAT]

# Expected format of line ending, e.g. empty (any line ending), LF or CRLF.
expected-line-ending-format=LF

# Number of spaces of indent required inside a hanging or continued line.
indent-after-paren=2

# String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1
# tab).
indent-string='  '

# Maximum number of characters on a single line.
max-line-length=80

# Maximum number of lines in a module
max-module-lines=1000

# Allow the body of a class to be on the same line as the declaration if body
# contains single statement.
single-line-class-stmt=yes

# Allow the body of an if to be on the same line as the test if there is no
# else.
single-line-if-stmt=yes


[CLASSES]

# List of method names used to declare (i.e. assign) instance attributes.
defining-attr-methods=__init__


[DESIGN]

# Maximum number of arguments for function / method
max-args=5

# Maximum number of attributes for a class (see R0902).
max-attributes=7

# Maximum number of boolean expressions in a if statement
max-bool-expr=5

# Maximum number of branch for function / method body
max-branches=12

# Maximum number of locals for function / method body
max-locals=15

# Maximum number of parents for a class (see R0901).
max-parents=7

# Maximum number of public methods for a class (see R0904).
max-public-methods=20

# Maximum number of return / yield for function / method body
max-returns=6

# Maximum number of statements in function / method body
max-statements=50

# Minimum number of public methods for a class (see R0903).
min-public-methods=2


[OVERRIDES]

# List of pylint default rules to override
disable=missing-final-newline
```

### Expected behavior

I would expect the Travis output to match the OSX output.

### pylint --version output

```
Yales-MacBook-Pro:cats yaledhlab$ pylint --version
No config file found, using default configuration
pylint 1.8.2,
astroid 1.6.1
Python 3.6.1 |Anaconda custom (x86_64)| (default, May 11 2017, 13:04:09)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]
```

I'd be grateful for any help others can offer on this!

**Comment by duhaime** [comment]
A workaround is to move the pylintrc file to the project root and not specify an `--rcfile` argument when evoking pylint. That said, I'd prefer to keep the pylintrc file out of the project root if possible...

**Comment by PCManticore** [comment]
The only difference I see is the lack of `Using config file ...` in the Travis CI output. Are you running with pylint 1.8 in CI?

**Comment by duhaime** [comment]
@PCManticore Yes exactly, I take it that line is printed on condition a config file is found / interpreted. The version of pylint in the requirements.txt file is: `pylint==1.8.2`

**Comment by PCManticore** [comment]
We would still emit a `No config file found` in case we don't find a config file, which suggests to me that somehow it already finds one config file somewhere, just not the one you'd want. You might also try with a SSH debug build and some breakpoints in pylint's config loader to see what it actually loaded.

**Comment by anselmos** [comment]
I've got the same strange behaviour - but it looks like it does not uses proper pylintrc file settings for disabling some of the errors. 
I've checked on `pylint==1.8.2` and `pylint==1.6.5` with `python==3.5.5` and `python=3.6.4` and no luck. 
Locally It uses pylintrc file normally and the output is different than on Travis
With `python==2.7.13` and `pylint==1.6.5` it worked on a different project, but this one is in python3.

Is this more related to Travis environment or more Pylint not using proper settings? (I'm more into the first option)
Thanks for any support in this matter.

**Comment by PCManticore** [comment]
@anselmos this is probably a Travis issue. To be honest, I'm not even sure this is running pylint 1.8. Now I notice that @duhaime 's output on travis has `Invalid constant name "app" (invalid-name)`. But since 1.8, we emit this message as in `Constant name "app" does not conform to snake-style`. Since `const-naming-style=snake_case` is taken in consideration only on Pylint 1.8, it makes sense to have this error on Travis. It would be great to have the travis output to validate that this is indeed the problem @duhaime

**Comment by anselmos** [comment]
@PCManticore  I've managed to fix the issue by manually adding `too-few-public-methods` to `.pylintrc` file. It was the one error that did not appear on my docker-container and appeared on Travis. Once I've added it to `.pylintrc` rc-file It stopped. But it was like using Pylint default config with only those settings that are in rc-file - which is how Pylint works. But thanks for your input on this matter :)
